src/Hint/Context.hs

```diff
@@ -197,14 +197,17 @@ isPhantomModule mn = do (as,zs) <- getPhantomModules
 loadModules :: MonadInterpreter m => [String] -> m ()
 loadModules fs = do -- first, unload everything, and do some clean-up
                     reset
-                    doLoad fs `catchIE` (\e -> reset >> throwM e)
+                    targets <- mapM (\f->runGhc2 GHC.guessTarget f Nothing) fs
+                    runGhc1 GHC.setTargets targets
+                    doLoad GHC.LoadAllTargets `catchIE` (\e -> reset >> throwM e)
 
-doLoad :: MonadInterpreter m => [String] -> m ()
-doLoad fs = mayFail $ do
-                   targets <- mapM (\f->runGhc2 GHC.guessTarget f Nothing) fs
-                   --
-                   runGhc1 GHC.setTargets targets
-                   res <- runGhc1 GHC.load GHC.LoadAllTargets
+reload :: MonadInterpreter m => m ()
```

Comment:
Since this will be exported, it will need documentation. See how it is done for `reset`.